### PR TITLE
Search for prisoner by number

### PR DESCRIPTION
**Pull request by novotnyjakub**
For the moment there's no validation on the search field and it searches using left and right wildcard on the `number` attribute of `Prisoner`.

### CODE REVIEW

File: app/controllers/prisoners_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class PrisonersController < ApplicationController
   before_action :set_prisoner, only: %i(show edit update)
 
   def index
+    @q = params[:q] || ''
+    @prisoners = Prisoner.where('number LIKE ?',  "%#{@q}%") if @q.present?
```
Comment (dan-palmer) [inline app/controllers/prisoners_controller.rb]: How about having this as a scope on `Prisoner`

File: config/locales/en.yml [hunk 2]
```diff
@@ -35,6 +35,13 @@ en:
         Enabling the safe, secure movement of people in custody
         and everyone involved.
       start_form: Add prisoner details
+      sections:
+        create:
+          heading: "Create new prisoner record"
```
Comment (dan-palmer) [inline config/locales/en.yml]: Can actually get away with not quoting the strings in locale file

File: spec/features/prisoner_search_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.feature 'Prisoner search', type: :feature do
```
Comment (dan-palmer) [inline spec/features/prisoner_search_spec.rb]: I'd be tempted to have this as:

```ruby
RSpec.feature 'User searches for prisoner by number', type: :feature do
  ...
```

& then just have the scenarios as:

```ruby
scenario 'when there are no results' do
  ...

scenario 'when there are results' do
  ..
```

File: app/controllers/prisoners_controller.rb (hunk 1 above)
Comment (dan-palmer) [inline app/controllers/prisoners_controller.rb]: Think we should go with exact matches for now as well, I feel as though partial matches add more complexity(e.g handling pagination)

I also worry that allowing 'lazy' searches that return multiple results might increase the chance of the wrong PER's being edited & I'd imagine the user performing the search would always have the full number in front of them. 

How about having this as a feature if user research dictates it as a requirement/user nicety?

**Comment by novotnyjakub** [comment]
Thanks for the comments, I made the suggested changes.

File: config/locales/en.yml [hunk 4]
```diff
@@ -35,6 +35,13 @@ en:
         Enabling the safe, secure movement of people in custody
         and everyone involved.
       start_form: Add prisoner details
+      sections:
+        create:
+          heading: Create new prisoner record
+        search:
+          heading: Prisoners search
+          results_heading: Results"
```
Comment (dan-palmer) [inline config/locales/en.yml]: Looks like one escaped 😁

File: spec/support/site_prism/pages/home.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+module Pages
+  class Home < SitePrism::Page
```
Comment (dan-palmer) [inline spec/support/site_prism/pages/home.rb]: This is cool :+1:

**Pull request merged**